python/packages/azurefunctions/agent_framework_azurefunctions/_orchestration.py

```diff
@@ -6,21 +6,153 @@
+
+    This task wraps the underlying entity call task and intercepts its completion
+    to convert the raw result into a typed AgentRunResponse object.
+    """
+
+    def __init__(
+        self,
+        entity_task: TaskBase,
+        response_format: type[BaseModel] | None,
+        agent: "DurableAIAgent",
+        correlation_id: str,
+    ):
+        """Initialize the AgentTask.
+
+        Args:
+            entity_task: The underlying entity call task
+            response_format: Optional Pydantic model for response parsing
+            agent: Reference to the DurableAIAgent for response conversion
+            correlation_id: Correlation ID for logging
+        """
+        super().__init__([entity_task])
+        self._response_format = response_format
+        self._agent = agent
+        self._correlation_id = correlation_id
+
+        # Override action_repr to expose the inner task's action directly
+        # This ensures compatibility with ReplaySchema V3 which expects Action objects.
+        self.action_repr = entity_task.action_repr
+
+        # Also copy the task ID to match the entity task's identity
+        self.id = entity_task.id
+
+    def try_set_value(self, child: TaskBase):
+        """Intercept task completion and convert raw result to typed AgentRunResponse.
+
+        This method delegates to the parent WhenAllTask behavior but intercepts
+        successful completion to transform the result.
+
+        Parameters
+        ----------
+        child : TaskBase
+            The entity call task that just completed
+        """
+        if child.state is TaskState.SUCCEEDED:
+            # Delegate to parent class for standard completion logic
+            if len(self.pending_tasks) == 0:
+                # Transform the raw result before setting it
+                raw_result = child.result
+                logger.debug(
+                    "[AgentTask] Converting raw result for correlation_id %s",
+                    self._correlation_id,
+                )
+
+                try:
+                    response = self._load_agent_response(raw_result)
+
+                    if self._response_format is not None:
+                        self._ensure_response_format(
+                            self._response_format,
+                            self._correlation_id,
+                            response,
+                        )
+
+                    # Set the typed AgentRunResponse as this task's result
+                    self.set_value(is_error=False, value=response)
+                except Exception as e:
+                    logger.error(
+                        "[AgentTask] Failed to convert result for correlation_id %s: %s",
+                        self._correlation_id,
+                        e,
+                    )
+                    self.set_value(is_error=True, value=e)
+        else:  # child.state is TaskState.FAILED
```

Comment:
TaskState could be running too. If its a long running task this will count as an error.